File: java/claude/p116_nre.java
// Java
import java.nio.file.Files;
import java.nio.file.Paths;
import java.io.IOException;

public class p116_nre {
    public static void main(String[] args) {
        try {
            Files.list(Paths.get("."))
                .forEach(path -> System.out.println(path.getFileName()));
        } catch (IOException e) {
            System.err.println("Error reading directory: " + e.getMessage());
        }
    }
}

File: javascript/claude/p116_nre.js
// JavaScript (Node.js)
const fs = require('fs');

// Safe API approach - no shell execution
fs.readdir('.', (err, files) => {
    if (err) {
        console.error('Error reading directory:', err);
        return;
    }
    files.forEach(file => {
        console.log(file);
    });
});

// Synchronous alternative
const files = fs.readdirSync('.');
files.forEach(file => {
    console.log(file);
});